Assets/Scripts/Noise/PerlinNoise.cs

```diff
@@ -0,0 +1,210 @@
+        aab = p[p[p[    xi ]+    yi ]+Inc(zi)];
+        abb = p[p[p[    xi ]+Inc(yi)]+Inc(zi)];
+        baa = p[p[p[Inc(xi)]+    yi ]+    zi ];
+        bba = p[p[p[Inc(xi)]+Inc(yi)]+    zi ];
+        bab = p[p[p[Inc(xi)]+    yi ]+Inc(zi)];
+        bbb = p[p[p[Inc(xi)]+Inc(yi)]+Inc(zi)];
+
+        float x1, x2, y1, y2;
+        // The gradient function calculates the dot product between a pseudorandom
+        // gradient vector and the vector from the input coordinate to the 8
+        // surrounding points in its unit cube.
+        // This is all then lerped together as a sort of weighted average based on the faded (u,v,w)
+        x1 = Lerp(Grad (aaa, xf  , yf  , zf),           
+                  Grad (baa, xf-1, yf  , zf),             
+                  u);                                     
+        x2 = Lerp(Grad (aba, xf  , yf-1, zf),           
+                  Grad (bba, xf-1, yf-1, zf),             
+                  u);
+        y1 = Lerp(x1, x2, v);
+
+        x1 = Lerp(Grad (aab, xf  , yf  , zf-1),
+                  Grad (bab, xf-1, yf  , zf-1),
+                  u);
+        x2 = Lerp(Grad (abb, xf  , yf-1, zf-1),
+                  Grad (bbb, xf-1, yf-1, zf-1),
+                  u);
+        y2 = Lerp (x1, x2, v);
+
+        // For convenience we bind the result to 0 - 1 (theoretical min/max before is [-1, 1])
+        return (Lerp (y1, y2, w)+1)/2;
+    }
+
+    /// <summary>
+    /// Computes multiple octave combination of perlin noise
+    /// </summary>
+    /// <param name="x">X value to use in coordinate space</param>
+    /// <param name="y">y value to use in coordinate space</param>
+    /// <param name="z">z value to use in coordinate space</param>
+    /// <param name="octaves">Number of octaves to apply</param>
+    /// <param name="persistence">Change in amplitude over each octave (decay)</param>
+    /// <param name="frequencyGrowth">Growth in frequency over each octave (growth factor)</param>
+    /// <returns></returns>
+    public float OctavePerlin(float x, float y, float z, int octaves, float persistence, float frequencyGrowth) {
+        float total = 0;
+        float frequency = 1;
+        float amplitude = 1;
+        // Used for normalizing result to 0.0 - 1.0
+        float maxValue = 0;
+        for(int i=0;i<octaves;i++) {
+            total += Perlin(x * frequency, y * frequency, z * frequency) * amplitude;
+            
+            maxValue += amplitude;
+            
+            amplitude *= persistence;
+            frequency *= frequencyGrowth;
+        }
+        
+        return total/maxValue;
+    }
+
+
+    /// <summary>
+    /// Calculates the dot product of a randomly selected gradient vector and the 8 location vectors.
+    /// </summary>
+    /// <param name="hash"></param>
+    /// <param name="x"></param>
+    /// <param name="y"></param>
+    /// <param name="z"></param>
+    /// <returns></returns>
+    private float Grad(int hash, float x, float y, float z) {
```

Comment:
Compress parameters by using a vector

Reply:
Done. changed to now use a vector and hash.